testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorTest.java

```diff
@@ -397,6 +398,111 @@ public void testDescribeTable()
         assertEquals(actualColumns, expectedColumns);
     }
 
+    @Override
+    public void testView()
+    {
+        // TODO merge AbstractTestDistributedQueries into BaseConnectorTest
+        super.testView();
+    }
+
+    @Test
+    public void testMaterializedView()
+    {
+        if (!hasBehavior(SUPPORTS_CREATE_MATERIALIZED_VIEW)) {
+            assertQueryFails("CREATE MATERIALIZED VIEW nation_mv AS SELECT * FROM nation", "This connector does not support creating materialized views");
+            return;
+        }
+
+        String catalogName = getSession().getCatalog().orElseThrow();
+        String schemaName = getSession().getSchema().orElseThrow();
+        String viewName = "test_materialized_view_" + randomTableSuffix();
+        assertUpdate("CREATE MATERIALIZED VIEW " + viewName + " AS SELECT * FROM nation");
+        // reading
+        assertThat(query("SELECT * FROM " + viewName))
+                .skippingTypesCheck()
+                .matches("SELECT * FROM nation");
+
+        // table listing
+        assertThat(query("SHOW TABLES"))
+                .skippingTypesCheck()
+                .containsAll("VALUES '" + viewName + "'");
+        // information_schema.columns without table_name filter
+        assertThat(query("SELECT table_name FROM information_schema.tables WHERE table_schema = '" + schemaName + "'"))
+                .skippingTypesCheck()
+                .containsAll("VALUES '" + viewName + "'");
+        // information_schema.columns with table_name filter
+        checkInformationSchemaTablesForPointedQueryForMaterializedView(schemaName, viewName);
+
+        // column listing
+        checkShowColumnsForMaterializedView(viewName);
+
+        // information_schema.columns without table_name filter
+        checkInformationSchemaColumnsForMaterializedView(schemaName, viewName);
+
+        // information_schema.columns with table_name filter
+        checkInformationSchemaColumnsForPointedQueryForMaterializedView(schemaName, viewName);
+
+        // view-specific listings
+        checkInformationSchemaViewsForMaterializedView(schemaName, viewName);
+
```

Comment:
We can also add system.jdbc tests similar to https://github.com/trinodb/trino/pull/8113/files#diff-6f73e014ee92e4e98ff14049eaa094585b893d44d7ee942a4c5a32deee15d836R129

Reply:
awesome, however, i would prefer to have this tested end-to-end via java.sql.DatabaseMetadata.
it then is kind of cumbersome as a connector test

@raunaqmorarka can you please file an issue to have DatabaseMetadata tested with MVs?

Reply:
created https://github.com/trinodb/trino/issues/8128